oidc-wp/openid-connect-generic
# feat(Cookies): Updates WP Cookie Expiration to Same as Session Length

### All Submissions:






### Changes proposed in this Pull Request:



Closes #512 .

This change allows the user to, using a filter, make the WP user session length the same as the refresh token expiration.
So if a refresh token is valid for 30 days, the wordpress login cookies will then be set for 30 days instead of the default of 14 days. Please note that the remember-me option needs to be enabled as well, as wordpress does not allow changing the cookie expiration if remember-me is not used.

Please note that this PR requires the code of PR #513 , and also includes that code for now untill PR #513 is merged.

### How to test the changes in this Pull Request:

1. In your theme, add `add_filter( 'openid-connect-generic-remember-me', '__return_true' );` to the functions.php file.
2. In your theme, add `add_filter( 'openid-connect-generic-use-token-refresh-expiration', '__return_true' );` to the functions.php file.
3. Login using the openid connect plugin
4. Open the development tools of your browser, and look at the cookie expiration. It should not be a session cookie, and it should be valid for the same time as the refresh token of the IDP.

### Other information:




### Changelog entry

> Added the `openid-connect-generic-use-token-refresh-expiration` filter to allow theme makers to login users with the same wp login cookie expiration time as the IDP.
